feat(challenge-header): api refact replace status submission with action (#888)

Ilmira83 opened the pull request:
**Summary**

This PR refactors the frontend submission flow so the client sends a user action (e.g. SAVE, SUBMIT, GIVE_UP) to the backend instead of attempting to send the resulting status. The backend becomes the single source of truth for the resulting status lifecycle (and returns the confirmed status in the response). This resolves FE/BE naming mismatches and removes hardcoded status values on the frontend.

**Main points:**

- SolutionService.submitSolution(...) now accepts action: string (or SolutionAction enum value) instead of status.
- The PUT request body contains action (no status field).
- Components (ChallengeHeaderComponent, SendSolutionModal, etc.) now pass actions ('SAVE', 'SUBMIT', 'GIVE_UP') when calling the service.
- Local UI state (this.solutionState) is updated only from the status returned by the backend response.
- i18n mappings updated where necessary to reflect BE status strings → localized labels.
- Unit tests updated to reflect the change.

🆕 Added SolutionAction Enum

- Created a new SolutionAction enum containing explicit user actions sent from the frontend to the backend (SAVE, SUBMIT, GIVE_UP).
- Updated all FE calls to use this enum when sending actions to the backend.
- Ensures a clean separation between actions (what the user does) and statuses (what the backend decides).

🧩 Retained Existing SolutionStatus Enum

- Kept the original SolutionStatus enum for internal FE state mapping and UI logic.
- FE now relies exclusively on the backend-returned status to update local solution state, removing any hardcoded FE logic.
- This preserves backward compatibility while aligning with the new BE contract.

**Tests**

- Update unit tests that previously asserted the request body had a status field to expect action (e.g. action: 'SAVE').
- Update tests that were passing SolutionStatus.* into submitSolution to pass SolutionAction.* (or string literals) instead.
- Ensure tests assert that this.solutionState (or equivalent) is set using the status from the mocked backend response, not a frontend-derived value.

eduzavarce committed:
fix: ensure full-height layout for main container and add regression tests

eduzavarce committed:
fix: ensure full-height layout for main container and add regression tests

eduzavarce committed:
fix: simplify test assertions in main component specs

eduzavarce committed:
fix: simplify test assertions in main component specs

eduzavarce committed:
fix: bump version to 3.11.5-RELEASE and document main container scrollbar fix in changelog

eduzavarce committed:
fix: bump version to 3.11.5-RELEASE and document main container scrollbar fix in changelog

eduzavarce committed:
feat: add collapsible filters UI for languages, levels, and progress

eduzavarce committed:
fix: adjust filters UI for improved alignment and responsiveness

- Updated HTML structure for consistent chevron alignment.
- Made layout adjustments to buttons using `inline-flex` for better sizing.
- Reduced chevron size and fine-tuned its rotation behavior.
- Adjusted desktop filters column to shrink-to-content for better responsiveness.

eduzavarce committed:
fix: refine filters UI spacing and remove redundant margin

- Reduced gap between title and chevron for better alignment.
- Removed unnecessary top margin from filters section title.

eduzavarce committed:
feat: implement "All" option for language tags and streamline filters UI

- Added "All" checkbox for selecting/deselecting all tags within a language.
- Updated filters logic to handle "All" selection state and synchronize with individual tags.
- Removed collapsible sections for levels and progress filters.
- Adjusted UI spacing, styles, and compacted checkboxes for a cleaner layout.
- Added translations for "All" in multiple languages.

eduzavarce committed:
feat: enhance progress filters with solution status mapping

- Added client-side filtering for challenges based on solution statuses (Not Started, In Progress, Finished).
- Updated logic to handle dynamic filters and sorting integration.
- Adjusted fetch behavior to ensure statuses are retrieved for admin users.
- Added comprehensive test coverage for progress filtering functionality.

eduzavarce committed:
fix: improve filters logic and clean up unused styles

- Replaced object key-based filtering with simpler `Object.values` iteration.
- Fixed inconsistent test sorting by introducing `localeCompare`.
- Removed redundant `#filters-container` styles for better maintainability.

eduzavarce committed:
fix: add spacing between filters and challenges for better UI layout

Ilmira83 committed:
fix: update FE naming to match backend

Ilmira83 committed:
fix: update FE naming to match backend

Ilmira83 committed:
fix: update FE naming to match backend

Ilmira83 committed:
fix: update i18n json files status name mapping

Ilmira83 committed:
fix: update i18n json files status name mapping

Ilmira83 committed:
fix: update i18n json files status name mapping

Ilmira83 committed:
fix: update i18n json files status name mapping

Ilmira83 committed:
feat:Add SolutionAction enum for API refactor

Ilmira83 committed:
feat:Add SolutionAction enum for API refactor

Ilmira83 committed:
feat:Add SolutionAction enum for API refactor

Ilmira83 committed:
fix: change method submitSolution. now accept action instead of status

Ilmira83 committed:
fix: change method submitSolution. now accept action instead of status

Ilmira83 committed:
fix: change method submitSolution. now accept action instead of status

Ilmira83 committed:
fix: updated acceptSolution to send SolutionAction instead of status field

Ilmira83 committed:
fix: updated acceptSolution to send SolutionAction instead of status field

Ilmira83 committed:
fix: updated acceptSolution to send SolutionAction instead of status field

Ilmira83 committed:
fix: updated saveChallenge to send SolutionAction instead of status field

Ilmira83 committed:
fix: updated saveChallenge to send SolutionAction instead of status field

Ilmira83 committed:
fix: updated saveChallenge to send SolutionAction instead of status field

Ilmira83 committed:
fix: updated saveChallenge to send SolutionAction instead of status field

Ilmira83 committed:
fix: update unit test for challenge-header-component

Ilmira83 committed:
fix: update unit test for challenge-header-component

Ilmira83 committed:
fix: update unit test for challenge-header-component

Ilmira83 committed:
fix: update unit test for send-solution-modal

Ilmira83 committed:
fix: update unit test for send-solution-modal

Ilmira83 committed:
fix: update unit test for send-solution-modal

Ilmira83 committed:
fix: update unit tests for SolutionService

Ilmira83 committed:
fix: update unit tests for SolutionService

Ilmira83 committed:
fix: update unit tests for SolutionService

Ilmira83 committed:
fix: update unit tests for SolutionService

Ilmira83 committed:
fix: update unit tests for SolutionService

Ilmira83 committed:
fix: sonar issues in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues p.2 in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues p.2 in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues p.2 in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues code duplications in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues code duplications in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues code duplications in solutionService.spec.ts

# Conflicts:
#	src/app/services/solution.service.spec.ts

Ilmira83 committed:
fix: sonar issues code duplications in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues code duplications in solutionService.spec.ts

# Conflicts:
#	src/app/services/solution.service.spec.ts

Ilmira83 committed:
fix: sonar issues code duplications in solutionService.spec.ts

Ilmira83 committed:
fix: sonar issues code duplications in solutionService.spec.ts

ivilarop committed:
Merge pull request #714 from IT-Academy-BCN/fix/876-challenge-list-scroll

fix(main-component): fix challenge list scroll (#876)

eduzavarce committed:
fix: make challenge filters collapsible and update version to 3.11.6-RELEASE

eduzavarce reviewed (approved):
Great job, all the requirements seem to be met.

sofrayala reviewed (approved):
Functionalities works as expected, looking goos. Congrats! :)

Ilmira83 committed:
update release version

Ilmira83 committed:
update release version

Ilmira83 committed:
update release version

Ilmira83 committed:
Merge branch 'feat/888-api_refact-replace-status-submission-with-action' of https://github.com/IT-Academy-BCN/ita-challenges-frontend into feat/888-api_refact-replace-status-submission-with-action

eduzavarce committed:
Merge branch 'develop' into feat/894-make_the_filters_panel_collapsible_and_hidden_by_default

ivilarop commented on conf/.env.CI.dev:
```diff
@@ -1,2 +1,2 @@
 MICROSERVICE_DEPLOY=ita-challenges-frontend
-MICROSERVICE_VERSION=3.11.4-RELEASE
+MICROSERVICE_VERSION=3.11.6-RELEASE
```
This clearly fits with a new MINOR, not a PATCH. 

Right now develop is at 3.11.4-RELEASE, so from this branch you should bump it to 3.12.0-RELEASE.
Each PR must bump the version starting from the one currently in develop, not from versions used in other branches/PRs. If another PR is merged first and changes the version, you’ll just update this value during rebase/merge.

Ilmira83 replied:
Release version changed to 3.12.0

Ilmira83 replied:
Updated to 3.13.0

ivilarop commented on src/app/core/layout/main/main.component.scss:
```diff
@@ -44,6 +44,8 @@
   }
   #main-container {
     overflow: auto;
+    display: flex;
```
Why is this layout change needed for this user story?
The PR is about replacing the submit status with an action, but this SCSS change modifies the global #main-container layout (display: flex; flex-direction: column;).
If it’s fixing a separate layout issue, it should probably go in its own PR (or at least be clearly documented in the summary), because it may have side effects on other pages.

Ilmira83 replied:
I was not asked and didn't make this changes, it was Eduardo's task, I will check with him. I did the rebase of the develop before pushing the updated RELEASE version to my branch after Eduardo's PR was merged, so I think that's why this is here now.

Ilmira83 replied:
We did the rebase develop / pull and push my branch again with Eduardo after his second PR was merged today and we didn't understand why all his commits appeared in my PR. I asked AI and it says that this can happen because GitHub shows those commits in my PR because the base comparison changed after the rebase, even though they are already part of develop. My branch itself doesn't include any changes from those files — it’s only a history visualization issue. So for my branch it’s only a PR history diff issue, not real changes.

ivilarop replied:
This happens because at some point your branch was based on Eduardo's branch, or you performed a merge/pull from it. That's why his commits are part of this pull request's history, and GitHub displays them, even though the file's contents are now the same as in the develop branch.

We'll leave it like this in this PR, but as a process improvement, from now on feature branches will always be created from develop and we will avoid basing them on other colleagues' branches so that their commits don't get mixed up in our PRs.

ivilarop commented on src/app/core/layout/main/main.component.spec.ts:
These new layout regression tests don’t seem related to the scope of this PR (replacing status with action).
If the SCSS change is required for this feature, please explain the functional need in the PR description.
Otherwise, both the style change and these tests should be moved to a separate PR to avoid mixing unrelated concerns and introducing unexpected UI side effects.

Ilmira83 replied:
See the previous comment.

ivilarop replied:
See the previous comment.

ivilarop commented on src/app/modules/challenge/components/challenge-header/challenge-header.component.spec.ts:
SolutionStatus.SUBMITTED_COMPLETE ?

Ilmira83 replied:
It should be SolutionStatus.ENDED here which is the name of the BE string value SUBMITTED_COMPLETE in the enum file. ENDED = "SUBMITTED_COMPLETE". I didn't change previous naming in order to escape more code refacturing and also to keep the name of the status (ENDED) different from the string value (SUBMITTED_COMPLETE) which we expect to get from BE.

ivilarop replied:
ok thks!

ivilarop commented on src/app/modules/challenge/components/challenge-header/challenge-header.component.spec.ts:
SolutionStatus.SUBMITTED_COMPLETE?

Ilmira83 replied:
It should be SolutionStatus.ENDED here which is the name of the BE string value SUBMITTED_COMPLETE in the enum file. ENDED = "SUBMITTED_COMPLETE". I didn't change previous naming in order to escape more code refacturing and also to keep the name of the status (ENDED) different from the string value (SUBMITTED_COMPLETE) which we expect to get from BE.

ivilarop replied:
ok thks!

ivilarop commented on src/app/modules/challenge/components/challenge-header/challenge-header.component.spec.ts:
SolutionStatus.SUBMITTED_COMPLETE?

Ilmira83 replied:
It should be SolutionStatus.ENDED here which is the name of the BE string value SUBMITTED_COMPLETE in the enum file. ENDED = "SUBMITTED_COMPLETE". I didn't change previous naming in order to escape more code refacturing and also to keep the name of the status (ENDED) different from the string value (SUBMITTED_COMPLETE) which we expect to get from BE.

ivilarop replied:
ok thks!

ivilarop commented on src/app/modules/challenge/components/challenge-header/challenge-header.component.ts:
SolutionStatus.SUBMITTED_COMPLETE?

Ilmira83 replied:
It should be SolutionStatus.ENDED here which is the name of the BE string value SUBMITTED_COMPLETE in the enum file. ENDED = "SUBMITTED_COMPLETE". I didn't change previous naming in order to escape more code refacturing and also to keep the name of the status (ENDED) different from the string value (SUBMITTED_COMPLETE) which we expect to get from BE.

ivilarop replied:
ok thks!

ivilarop commented on src/app/modules/challenge/components/challenge-header/challenge-header.component.ts:
SolutionStatus.SUBMITTED_COMPLETE?

Ilmira83 replied:
It should be SolutionStatus.ENDED here which is the name of the BE string value SUBMITTED_COMPLETE in the enum file. ENDED = "SUBMITTED_COMPLETE". I didn't change previous naming in order to escape more code refacturing and also to keep the name of the status (ENDED) different from the string value (SUBMITTED_COMPLETE) which we expect to get from BE.

ivilarop replied:
ok thks!

ivilarop commented on src/app/modules/challenge/components/challenge-header/challenge-header.component.ts:
```diff
@@ -182,28 +184,27 @@ export class ChallengeHeaderComponent implements OnInit {
+      this.solutionText
+    ).subscribe({
+      next: () => {
+        this.solutionState = SolutionStatus.IN_PROGRESS; // Hardcoded status value. Need to be called by fetchUserSolution?
```
The backend returns the state in the submitSolution response.

next: (response) => {
  this.solutionState = response.status as SolutionStatus;
}

ivilarop replied:
Yes, I see that this part hasn't been modified, but can we take advantage of this task to avoid encoding the value and retrieve it from the previous response?

Ilmira83 replied:
Done. But in this method this single refacture has no sense in general. The method need to be supplemented with other functionalities in order to handle properly further state and UI changing. But this is out of scope of my current task. Need to reflect this in Issues.

ivilarop commented on CHANGELOG.md:
```diff
@@ -3,6 +3,15 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+### [ita-challenges-frontend-3.11.6-RELEASE] - 2025-11-25
```
In the same HU, two versions have appeared in the changelog (3.11.5 and 3.11.6) and that shouldn't happen.

I don't understand the change: - Fixed Main container size adjusted to hold all the screen, resolving issues with challenge list scrollbars (Taiga[#876],PR [#714]). I don't see task 876 and PR 714 doesn't exist. Did Luis request this change? It's not part of the HU

Ilmira83 replied:
I didn't make this changes, it was Eduardo's task, I will check with him. I did the rebase of the develop before pushing the updated RELEASE version to my branch after Eduardo's PR was merged, so I think that's why this is here now.

Ilmira83 replied:
The version updated up to 3.13.0. For the rest - please see my comment above regarding rebase issues.

ivilarop commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ita-challenges-frontend",
-  "version": "3.11.4-RELEASE",
+  "version": "3.11.6-RELEASE",
```
see previous comments

Ilmira83 replied:
Release version changed to 3.12.0

Ilmira83 replied:
Upd to 3.13.0

eduzavarce committed:
fix: update progress filters to use `SolutionStatus` enum and optimize admin logic

- Replaced hardcoded progress filter values with `SolutionStatus` enum for better maintainability.
- Refined admin logic to skip unnecessary `fetchUserSolutionsStatus` call when user is admin.

eduzavarce committed:
bump version to 3.12.0-RELEASE and update changelog

Ilmira83 committed:
update RELEASE version to  3.12.0

Ilmira83 committed:
update RELEASE version to  3.12.0

eduzavarce committed:
refactor: replace `SolutionStatus` with `ProgressFilter` enum in progress filters logic

- Introduced new `ProgressFilter` enum for improved clarity and maintainability.
- Updated filtering logic and test cases to use `ProgressFilter` instead of `SolutionStatus`.

eduzavarce committed:
fix: update test to avoid calling `fetchUserSolutionsStatus` on init for admin users

eduzavarce committed:
fix: refactor solution service tests to remove redundant `done` callbacks and streamline assertions

eduzavarce committed:
fix: refactor solution service tests to remove redundant `done` callbacks and streamline assertions

eduzavarce committed:
fix: refactor solution service tests to remove redundant `done` callbacks and streamline assertions

eduzavarce committed:
fix: remove redundant test case and unused imports in solution service specs

eduzavarce committed:
fix: remove redundant test case and unused imports in solution service specs

eduzavarce committed:
refactor: remove `ProgressFilter` enum and update progress logic to use `SolutionStatus`

- Eliminated `ProgressFilter` enum in favor of `SolutionStatus` for consistency.
- Updated filtering logic and test cases to reflect the change.

ivilarop committed:
Merge pull request #715 from IT-Academy-BCN/feat/894-make_the_filters_panel_collapsible_and_hidden_by_default

feat(starter-filters) make the filters panel collapsible and hidden by default (#894)

Ilmira83 committed:
upd: update status from response in saveChallenge method

Ilmira83 committed:
upd: update status from response in saveChallenge method

Ilmira83 committed:
Merge branch 'feat/888-api_refact-replace-status-submission-with-action' of https://github.com/IT-Academy-BCN/ita-challenges-frontend into feat/888-api_refact-replace-status-submission-with-action

Ilmira83 committed:
release version update

Ilmira83 committed:
Merge branch 'feat/888-api_refact-replace-status-submission-with-action' of https://github.com/IT-Academy-BCN/ita-challenges-frontend into feat/888-api_refact-replace-status-submission-with-action

Ilmira83 commented on CHANGELOG.md:
```diff
@@ -3,6 +3,20 @@
 and this project adheres to
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+### [ita-challenges-frontend-3.13.0-RELEASE] - 2025-11-27
+
```
Updated to version 3.13.0.

Ilmira83 commented on src/app/core/layout/main/main.component.spec.ts:
```diff
@@ -39,4 +42,32 @@ describe('MainComponent', () => {
+
+    expect(scss).toContain('#main-wrapper')
+    expect(scss).toMatch(/#main-wrapper[\s\S]*height:\s*100%/)
+  })
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

ivilarop replied:
This file (main.component.spec.ts) and the SCSS changes come from Eduardo’s PR #715, because the feature branch was created or rebased from a branch that already included those commits. They are not actual changes from this user story, just history diff noise.

We have verified that the final content matches develop, so they do not introduce any functional changes. It's safe to merge, but for the next PRs please ensure feature branches are created directly from develop to avoid mixing history from other user stories.

Ilmira83 commented on src/app/models/filter-challenge.model.ts:
```diff
@@ -1,6 +1,8 @@
   levels: string[] = []
-  progress: number[] = []
+  progress: SolutionStatus[] = []
   tags?: string[] = []
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

ivilarop replied:
These changes in filter-challenge.model.ts come from Eduardo's PR #715.
The feature branch originally contained the historical commit where this type change was introduced.

I have verified with git diff origin/develop -- src/app/models/filter-challenge.model.ts that the file is identical to develop, so this is not a real change from this user story — just a history diff caused by how the branch was created or rebased.

It is safe to merge.

For the next PRs, please ensure feature branches are created directly from develop to avoid pulling historical commits from other user stories.

Ilmira83 commented on src/app/modules/modals/filters-modal/filters-modal.component.spec.ts:
```diff
@@ -70,7 +71,7 @@ describe('FiltersModalComponent', () => {
       languages: ['1'], // Solo JavaScript seleccionado
       levels: ['EASY'],
-      progress: [1] // noStarted corresponde al primer valor
+      progress: [SolutionStatus.NOT_STARTED] // noStarted corresponde
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

ivilarop replied:
We've reviewed the files listed as changes from Eduardo's PR #715 (main.component.scss, main.component.spec.ts, filter-challenge.model.ts, filters-modal.component.spec.ts, etc.).

In all cases, I've verified with `git diff origin/develop -- <file>` that the current content is identical to `develop`, so these aren't actual changes from this build but rather historical commits that were carried over when creating/developing the branch from another branch that already contained Eduardo's changes.

From a functional standpoint, merging is safe. For future builds, feature branches should always be pulled directly from `develop` to avoid carrying over commits from other builds.

Ilmira83 commented on src/app/modules/modals/filters-modal/filters-modal.component.ts:
```diff
@@ -79,9 +80,14 @@ export class FiltersModalComponent {
+      Object.entries(formValue.progress).forEach(([key, val]) => {
         if (val) {
-          filters.progress.push(i + 1)
+          filters.progress.push(progressMap[key])
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

ivilarop replied:
This file contains logic from Eduardo’s PR #715 (the progressMap refactor).
These commits were already in the feature branch history, which is why GitHub shows them here.
I have checked with git diff origin/develop that the final content matches develop, so this is not a real change from this user story.
It’s safe to merge.
For future branches, please create feature branches directly from develop to avoid inheriting commits from other PRs.

Ilmira83 commented on src/app/modules/starter/components/starter-filters/starter-filters.component.html:
```diff
@@ -17,7 +17,15 @@ <h3 class="font-weight-bold mb-2">
+          [id]="'toggle-' + langKey"
+          (click)="toggleLanguage(langKey)"
+        >
           <span class="lang-dot" *ngIf="hasSelectedTags(langKey)" aria-hidden="true"></span>
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

ivilarop replied:
Many files shown in this PR contain changes originally introduced in Eduardo’s PR #715
(example: accessibility refactor in starter-filters.component.html, progress mapping, layout updates, etc.).

The reason they appear here is that this feature branch was created or rebased from a branch that already contained Eduardo’s commits.

I verified using:

git diff origin/develop -- <file>


that the final content of all these files is identical to develop, so they are not part of this user story.

✔️ It is safe to merge.

For future work, please ensure feature branches are always created directly from develop to avoid inheriting commits from other PRs.

Ilmira83 commented on src/app/modules/starter/components/starter-filters/starter-filters.component.html:
```diff
@@ -26,62 +34,94 @@ <h3 class="font-weight-bold mb-2">
+                <label class="form-check-label" [for]="langKey + '-tag-all'">
+                  {{ 'common.all' | translate }}
+                </label>
+              </div>
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

ivilarop replied:
see previous comments

Ilmira83 commented on src/app/modules/starter/components/starter-filters/starter-filters.component.html:
```diff
@@ -26,62 +34,94 @@ <h3 class="font-weight-bold mb-2">
+                </label>
+              </div>
+              }
+            </fieldset>
```
This is again the changes of Eduardo's PR #715  . I didn't make them.

ivilarop replied:
see previous comments

Ilmira83 commented on src/app/modules/starter/components/starter-filters/starter-filters.component.html:
```diff
@@ -26,62 +34,94 @@ <h3 class="font-weight-bold mb-2">
+      <div class="form-check">
+        <input class="form-check-input" type="checkbox" id="checkHard" formControlName="hard">
+        <label class="form-check-label" for="checkHard"><span>{{'modules.starter.filters.hard' | translate}}</span></label>
+      </div>
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

ivilarop replied:
see previous comments

Ilmira83 commented on src/app/modules/starter/components/starter-filters/starter-filters.component.html:
```diff
@@ -26,62 +34,94 @@ <h3 class="font-weight-bold mb-2">
+      <div class="form-check">
+        <input class="form-check-input" type="checkbox" id="checkFinished" formControlName="finished">
+        <label class="form-check-label" for="checkFinished"><span>{{'modules.starter.filters.finished' | translate}}</span></label>
+      </div>
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

ivilarop replied:
see previous comments

Ilmira83 commented on src/app/modules/starter/components/starter-filters/starter-filters.component.scss:
```diff
@@ -1,26 +1,40 @@
+        &:hover {
+            border-color: $primary;
+        }
         &:checked {
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

ivilarop replied:
see previous comments

Ilmira83 commented on src/app/modules/starter/components/starter-filters/starter-filters.component.scss:
```diff
@@ -81,5 +95,62 @@
+
+.collapsible-panel {
+  overflow: hidden;
+  transition: max-height 220ms ease-in-out, opacity 220ms ease-in-out;
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

ivilarop replied:
see previous comments

Ilmira83 commented on src/app/modules/starter/components/starter-filters/starter-filters.component.spec.ts:
```diff
@@ -151,6 +179,70 @@ describe('StarterFiltersComponent', () => {
+    expect(lastCallArgs.tags).toEqual([]);
+  });
+
   it('should toggle JS tag checkboxes and reflect the form state', async () => {
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

Ilmira83 commented on src/app/modules/starter/components/starter-filters/starter-filters.component.ts:
```diff
@@ -239,8 +232,91 @@ export class StarterFiltersComponent implements OnInit {
+      .forEach(([k, v]) => v && progressMap[k] && filters.progress.push(progressMap[k]));
+
+    this.filtersSelected.emit(filters);
+  }
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

Ilmira83 commented on src/app/modules/starter/components/starter/starter.component.html:
```diff
@@ -2,7 +2,10 @@
+    <div id="filters-panel" class="filters-panel open">
+      <app-starter-filters (filtersSelected)="getChallengeFilters($event)"></app-starter-filters>
+    </div>
   </div>
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

Ilmira83 commented on src/app/modules/starter/components/starter/starter.component.scss:
```diff
@@ -122,3 +132,94 @@ app-pagination {
+    border-radius: 50%;
+    background-color: $primary;
+  }
+}
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

Ilmira83 commented on src/app/modules/starter/components/starter/starter.component.spec.ts:
```diff
@@ -204,3 +222,77 @@ describe('StarterComponent', () => {
+      .sort((a: string, b: string) => a.localeCompare(b))
+    ).toEqual(['c1', 'c3'])
+  });
+});
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

Ilmira83 commented on src/app/modules/starter/components/starter/starter.component.ts:
```diff
@@ -147,14 +154,34 @@ export class StarterComponent implements OnInit {
+    })
   }
 
   changeSort(newSort: string): void {
```
This is again the changes of Eduardo's PR https://github.com/IT-Academy-BCN/ita-challenges-frontend/pull/715 . I didn't make them.

Ilmira83 commented on src/assets/i18n/ca.json:
```diff
@@ -196,7 +196,10 @@
         "IN_PROGRESS": "En progrés",
-        "ENDED": "Completat"
+        "SUBMITTED_COMPLETE": "Completat"
+    },
```
I changed only line 199.

Ilmira83 commented on src/assets/i18n/en.json:
```diff
@@ -197,7 +197,10 @@
         "IN_PROGRESS": "In progress",
-        "ENDED": "Completed"
+        "SUBMITTED_COMPLETE": "Completed"
+    },
```
I changed only line 200.

Ilmira83 commented on src/assets/i18n/es.json:
```diff
@@ -196,7 +196,10 @@
         "IN_PROGRESS": "En progreso",
-        "ENDED": "Completado"
+        "SUBMITTED_COMPLETE": "Completado"
+    },
```
I changed only line 199.

ivilarop committed:
Merge branch 'feat/888-api_refact-replace-status-submission-with-action' of https://github.com/IT-Academy-BCN/ita-challenges-frontend into feat/888-api_refact-replace-status-submission-with-action
